src/main/java/uk/gov/ons/ctp/common/event/EventPublisher.java

```diff
@@ -0,0 +1,16 @@
+package uk.gov.ons.ctp.common.event;
+
+import uk.gov.ons.ctp.common.error.CTPException;
+
+public interface EventPublisher {
+
+  /**
+   * Method to publish an asynchronous event message.
+   *
+   * @param routingKey message routing key
+   * @param payload message payload to send
+   * @return String of transaction Id
+   * @throws CTPException if cannot deal with payload
+   */
+  <T> String sendEvent(String routingKey, T payload) throws CTPException;
```

Comment:
Is it really worth having an interface? 
In this case it looks to add a little bit of bulk without enough corresponding benefit.

Reply:
Interface removed.